### PR TITLE
add gdal_config_options that are set as thread local

### PR DESCRIPTION
**Pull request by michaelmattig**

### CODE REVIEW

**Comment by michaelmattig** [comment]
[Agribora: Sentinel-1-AWS-Bucket anbinden](https://trello.com/c/8zXrfZmY/350-agribora-sentinel-1-aws-bucket-anbinden)

File: operators/src/source/gdal_source.rs [hunk 1]
```diff
@@ -154,7 +155,54 @@ pub struct GdalDatasetParameters {
...
+    // Dataset open option as strings, e.g. `vec!["UserPwd=geoengine:pwd".to_owned(), "HttpAuth=BASIC".to_owned()]`
     pub gdal_open_options: Option<Vec<String>>,
+    // Configs as key, value pairs that will be set as thread local config options, e.g.
+    // `vec!["AWS_REGION".to_owned(), "eu-central-1".to_owned()]` and unset afterwards
+    // TODO: validate the config options: only allow specific keys and specific values
+    pub gdal_config_options: Option<Vec<(String, String)>>,
+}
+
+/// Set thread local gdal options and revert them on drop
+struct GdalThreadLocalConfigOptions {
```
Comment (ChristianBeilschmidt) [inline operators/src/source/gdal_source.rs:167]:
```suggestion
struct TemporaryGdalThreadLocalConfigOptions {
```

File: operators/src/source/gdal_source.rs [hunk 2]
```diff
@@ -405,6 +454,13 @@ where
             .as_ref()
             .map(|o| o.iter().map(String::as_str).collect::<Vec<_>>());
 
+        // reverts the thread local configs on drop
+        info!("{:?}", dataset_params.gdal_config_options);
```
Comment (ChristianBeilschmidt) [inline operators/src/source/gdal_source.rs:458]: Don't you `info!` right within calling `new`? Isn't it doubled?

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]
looks solid

File: operators/src/source/gdal_source.rs (hunk 1 above)
Comment (michaelmattig) [reply operators/src/source/gdal_source.rs]: 6aedc4ecc6cbc634b02eb93affae62211749444c

File: operators/src/source/gdal_source.rs (hunk 2 above)
Comment (michaelmattig) [reply operators/src/source/gdal_source.rs]: 6aedc4ecc6cbc634b02eb93affae62211749444c

**Review by ChristianBeilschmidt — APPROVED** [review]

**Pull request merged**